terraform/executeTerraformTest.sh

```diff
@@ -62,37 +74,32 @@ case ${AWS_REGION} in
 esac
 
 test_framework_shortsha=$(git rev-parse --short HEAD)
-checkCache(){
-    CACHE_HIT=$(aws dynamodb get-item --region=us-west-2 --table-name ${DDB_TABLE_NAME} --key {\"TestId\":{\"S\":\"$1$2$3\"}\,\"aoc_version\":{\"S\":\"${TF_VAR_aoc_version}$test_framework_shortsha\"}})
-}
-
-checkCache $1 $2 $3
 # Used as a retry mechanic.
 ATTEMPTS_LEFT=2
 cd ${TEST_FOLDER};
-while [ $ATTEMPTS_LEFT -gt 0 ] && [ -z "${CACHE_HIT}" ]; do
+while [ $ATTEMPTS_LEFT -gt 0 ] && ! ../checkCacheHit.sh $SERVICE $TESTCASE $ADDTL_PARAMS; do
     terraform init;
-    if timeout -k 5m --signal=SIGINT -v 45m terraform apply -auto-approve -lock=false $opts  -var="testcase=../testcases/$2" ; then
+    if timeout -k 5m --signal=SIGINT -v 45m terraform apply -auto-approve -lock=false $opts  -var="testcase=../testcases/$TESTCASE" ; then
         APPLY_EXIT=$?
         echo "Exit code: $?" 
-        aws dynamodb put-item --region=us-west-2 --table-name ${DDB_TABLE_NAME} --item {\"TestId\":{\"S\":\"$1$2$3\"}\,\"aoc_version\":{\"S\":\"${TF_VAR_aoc_version}$test_framework_shortsha\"}\,\"TimeToExist\":{\"N\":\"${TTL_DATE}\"}} --return-consumed-capacity TOTAL
+        aws dynamodb put-item --region=us-west-2 --table-name ${DDB_TABLE_NAME} --item {\"TestId\":{\"S\":\"$SERVICE$TESTCASE$ADDTL_PARAMS\"}\,\"aoc_version\":{\"S\":\"$DDB_SK_PREFIX$test_framework_shortsha\"}\,\"TimeToExist\":{\"N\":\"${TTL_DATE}\"}} --return-consumed-capacity TOTAL
     else
         APPLY_EXIT=$?
         echo "Terraform apply failed"
         echo "Exit code: $?"
-        echo "AWS_service: $1"
-        echo "Testcase: $2" 
+        echo "AWS_service: $SERVICE"
+        echo "Testcase: $TESTCASE" 
     fi
 
-    case "$service" in
+    case "$SERVICE" in
         EKS*) terraform destroy --auto-approve $opts;
         ;;
     *)
         terraform destroy --auto-approve;
     ;;
     esac
 
-    checkCache $1 $2 $3 
+    ../checkCacheHit.sh $SERVICE $TESTCASE $ADDTL_PARAMS
```

Comment:
I think we can remove this line as you updated the condition to exit the loop so that it does not rely on CACHE_HIT:
```while [ $ATTEMPTS_LEFT -gt 0 ] && ! ../checkCacheHit.sh $SERVICE $TESTCASE $ADDTL_PARAMS; do```

Reply:
Good point! Nice catch.